onnxruntime/core/session/inference_session_utils.cc

```diff
@@ -227,6 +234,32 @@ Status JsonConfigParser::ParseRunOptionsFromModelProto(RunOptions& /*run_options
                          "Parsing RunOptions from ModelProto is not supported yet");
 }
 
+Status ParseTuningResultsFromModelMetadata(const onnxruntime::ModelMetadata& metadata,
+                                           std::vector<TuningResults>& results,
+                                           bool& key_found) {
+  results.clear();
+  key_found = false;
+  auto it = metadata.custom_metadata_map.find(kTuningResultsKeys);
+  if (it == metadata.custom_metadata_map.end()) {
+    return Status::OK();
+  }
+
+  key_found = true;
+  LOGS_DEFAULT(INFO) << "Found tuning results in the model file to be used while running the model";
+
+  ORT_TRY {
+    auto parsed_tuning_results_json = json::parse(it->second);
+    results = parsed_tuning_results_json.get<std::vector<TuningResults>>();
+  }
+  ORT_CATCH(const std::exception& e) {
```

Comment:
I think we need to also use ORT_HANDLE_EXCEPTION() for this to work in a build with exceptions disabled. e.g.:
https://github.com/microsoft/onnxruntime/blob/37033975bbc8fee7f32073b217654f308c529ccd/onnxruntime/core/session/inference_session.cc#L1529-L1534

on the other hand, since it is not enabled in a minimal build and I believe we only have CI builds which disable exceptions for minimal builds, perhaps we could just use `try/catch` directly.

Reply:
Updated to use `ORT_HANDLE_EXCEPTION`